test/input_gen/recorder.py

```diff
@@ -139,10 +142,10 @@ def train_step(model, optimizer, loss_fn, initial_input, label, writer_fn, **kwa
 
     # if traversal order is different from nntrainer, this need to be restructured to rely on the order of output.
     for layer in model.layers:
-        # we might need a bit of reordering if output is more than one
+        if isinstance(layer, K.layers.Permute):
+            continue
```

Comment:
Could you check if this line is used? I have made a patch which removes this line, maybe it got into from rebasing?

Reply:
done.